src/Confluent.Kafka/Producer.cs

```diff
@@ -510,13 +551,37 @@ public void HandleDeliveryReport(Message message)
             }
         }
 
-        public Task<Message<TKey, TValue>> ProduceAsync(string topic, TKey key, TValue val, DateTime? timestamp = null, int? partition = null, bool blockIfQueueFull = true)
+        private Task<Message<TKey, TValue>> Produce(string topic, TKey key, TValue val, DateTime? timestamp, int partition, bool blockIfQueueFull)
         {
             var handler = new TypedTaskDeliveryHandlerShim(key, val);
-            producer.ProduceAsync(topic, KeySerializer?.Serialize(key), ValueSerializer?.Serialize(val), handler, timestamp, partition, blockIfQueueFull);
+            var keyBytes = KeySerializer?.Serialize(key);
```

Comment:
Sorry for chiming in so late, but this statements seems like a bug to me. The `keyBytes` will be set to null when the key serializer is `null`. Ditto for `valBytes`. Is this intentional? If the serializer cannot ever be set to null, I think the elvis operator should be removed.